htmresearch/support/text_preprocess.py

```diff
@@ -68,12 +68,20 @@ def __init__(self,
 
   def _setupCorpus(self, corpusSource):
     """Create member vars for English language corpus and bag of words."""
-    corpusPath = os.path.abspath(os.path.join(
-      os.path.dirname(__file__), "../..", "projects/nlp/data/etc", corpusSource))
     try:
+      if os.path.isfile(corpusSource):
+        # absolute path
+        corpusPath = corpusSource
+      else:
+        # relative path
+        corpusPath = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                  "../../projects/nlp/data/etc",
```

Comment:
Not sure if we really care but this code is not portable to Windows. In order to be portable it should be something like:

```
os.path.join(os.path.dirname(__file__),
                                                  "..","..","projects/nlp/data/etc", corpusSource)
```